src/coaster_gui.cpp

```diff
@@ -126,21 +165,67 @@ CoasterInstanceWindow::CoasterInstanceWindow(CoasterInstance *ci) : GuiWindow(WC
 {
 	this->ci = ci;
 	this->SetupWidgetTree(_coaster_instance_gui_parts, lengthof(_coaster_instance_gui_parts));
+	this->SetCoasterState();
+	this->UpdateRecolourButtons();
+
+	entrance_exit_placement.cur_cursor = CUR_TYPE_INVALID;
+	/* When opening the window of a newly built ride immediately prompt the user to place the entrance or exit. */
+	if (this->ci->NeedsEntrance()) {
+		ChooseEntranceExitClicked(true);
+	} else if (this->ci->NeedsExit()) {
+		ChooseEntranceExitClicked(false);
+	}
 }
 
 CoasterInstanceWindow::~CoasterInstanceWindow()
 {
+	SetSelector(nullptr);
 	if (!GetWindowByType(WC_COASTER_BUILD, this->wnumber) && !this->ci->IsAccessible()) {
 		_rides_manager.DeleteInstance(this->ci->GetIndex());
 	}
 }
 
+assert_compile(MAX_RECOLOUR >= 3); ///< Check that the 3 recolourings of a gentle/thrill ride fit in the Recolouring::entries array.
+
+/** Update all recolour buttons of the window. */
+void CoasterInstanceWindow::UpdateRecolourButtons()
+{
+	for (int i = 0; i < 3; i++) {
+		const RecolourEntry &re = this->ci->entrance_recolours.entries[i];
+		this->GetWidget<LeafWidget>(CIW_ENTRANCE_RECOLOUR1 + i)->SetShaded(!re.IsValid());
+	}
+	for (int i = 0; i < 3; i++) {
```

Comment:
would a constant for the magic `3` make sense? At least it simplifies finding back the related magic numbers.